### PR TITLE
fix(ui): Make renaming widget easier

### LINKED ISSUE #44792: Not obvious how to rename a widget
### Problem Statement

image

image

### Solution Brainstorm

Literally could not find it. Had to have someone at Sentry show me. Just make it a form control?

### PR DESCRIPTION
**Pull request by om-ray**
This PR adds an input to rename widgets when editing.

Fixes GH-44792

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

**Comment by om-ray** [comment]
@dcramer @evanpurkhiser Please review this PR! Thank you!

**Comment by evanpurkhiser** [comment]
Thanks for opening this PR @om-ray. I'll route it over to those responsible for our dashboards + widgets :-) They probably won't get to it until Monday though, since they're based in Toronto .

**Comment by narsaynorath** [comment]
Hey again @om-ray! Sorry I haven't been able to review your PR, but I'll be able to take a look tomorrow at the start of my day. If you have some time before then, it looks like there are some tests that you may need to modify to work with the new flow that you're adding.

**Review by narsaynorath — COMMENTED** [review]
This is looking good, the only thing that looked off to me was the styling around the input field. My suggestion is just to remove the padding instead of applying a negative margin

Screenshot 2023-05-17 at 10 45 10 AM

Also, good job fixing the acceptance tests but there are still some jest tests failing. In particular these files:
- static/app/views/dashboards/widgetBuilder/widgetBuilderDataset.spec.tsx
- static/app/views/dashboards/widgetBuilder/widgetBuilder.spec.tsx

You can run the tests in these files by using `yarn test <path_to_the_test_file>` and you can zero in on specific tests by changing `it(...)` to `it.only(...)`. I believe you should be able to see if you click on the "Details" link next to the failed job for `frontend / Jest (0) (pull_request)` and `frontend / Jest (1) (pull_request)` to see which tests you may want to focus on. I looked at some of the test failures and it seems like it's just because we've swapped the component out for an input field and we can't do a text selection the same way anymore.

I think you'll need to apply the following change, with maybe one or two variations:

Before:
```js
        expect(await screen.findAllByText('Custom Widget')).toHaveLength(2);
```

After:
```js
        expect(await screen.findByRole('textbox', {name: /widget title/i})).toHaveValue(
          'Custom Widget'
        );
        expect(await screen.getByText('Custom Widget')).toBeInTheDocument();
```

The first one checks that the input field for the title has the expected value, and the other one matches the widget header text

**Comment by om-ray** [comment]
@narsaynorath Got it! Ill add these ASAP.

**Comment by narsaynorath** [comment]
Feel free to re-request me when your new changes are up :)

File: static/app/views/dashboards/widgetBuilder/widgetBuilder.tsx [hunk 1]
```diff
@@ -1138,6 +1149,10 @@ function WidgetBuilder({
 
 export default withPageFilters(withTags(WidgetBuilder));
 
+const TitleInput = styled(InputField)`
+  padding: 0;
```
Comment (narsaynorath) [inline static/app/views/dashboards/widgetBuilder/widgetBuilder.tsx:1153]: I know I already commented on a change here, but I realized that the right side should have a bit of padding to align the elements :)

Can we change this to:

```suggestion
  padding: 0 ${space(2)} 0 0;
```

**Review by narsaynorath — COMMENTED** [review]
There's another failing test in `widgetBuilder.spec.tsx` called `can respond to validation feedback`, can you replace it with this? This should test your new functionality.

```ts
  it('can respond to validation feedback', async function () {
    MockApiClient.addMockResponse({
      url: '/organizations/org-slug/dashboards/widgets/',
      method: 'POST',
      statusCode: 400,
      body: {
        title: ['This field may not be blank.'],
      },
    });

    renderTestComponent();

    await userEvent.click(await screen.findByText('Table'));

    const customWidgetLabels = await screen.findByText('Custom Widget');
    // EditableText and chart title
    expect(customWidgetLabels).toBeInTheDocument();

    await userEvent.clear(screen.getByRole('textbox', {name: 'Widget title'}));

    await userEvent.click(screen.getByText('Add Widget'));

    await screen.findByText('This field may not be blank.';
  });
```

**Comment by om-ray** [comment]
@narsaynorath Thanks for all the help. Sorry for the slow progress with this, I've been sick this past week, and whenever I run the tests on my local version, they seem to pass but then fail here. I do think I've fixed all the problems with the tests now, thanks to your help! Thanks for reviewing my changes and for all the patience!

**Review by narsaynorath — APPROVED** [review]
Sorry to hear that you were feeling sick, but the progress wasn't slow at all! Thanks again for your contribution, the changes look good and should help make renaming widgets a lot clearer

**Pull request merged**